### PR TITLE
Modis

### PR DESCRIPTION
**Pull request by davidcordeiro-deimos**
Fixes #

### Proposed fixes:
New harvester for MODIS products (Data Source: NASA CMR). This harvester allows to collect 12 new collections:

- MODIS_AQUA_MYD13Q1  (currently 110551 datasets, from 2002-07-04 to present - approximately 300 datasets every 16 days)
- MODIS_AQUA_MYD13A1  (currently 110551 datasets, from 2002-07-04 to present - approximately 300 datasets every 16 days)
- MODIS_AQUA_MYD13A2  (currently 110540 datasets, from 2002-07-04 to present - approximately 300 datasets every 16 days)
- MODIS_TERRA_MOD13Q1  (currently 126268 datasets, from 2000-02-18 to present - approximately 300 datasets every 16 days)
- MODIS_TERRA_MOD13A1  (currently 126268 datasets, from 2000-02-18 to present - approximately 300 datasets every 16 days)
- MODIS_TERRA_MOD13A2  (currently 126268 datasets, from 2000-02-18 to present - approximately 300 datasets every 16 days)
- MOD17A3H (currently   4110 datasets, from 2000-12-26 to 2015-01 - approximately 300 datasets every 01 year)
- MOD17A2H (currently 249848 datasets, from 2000-02-18 to present - approximately 300 datasets every 08 days)
- MODIS_AQUA_MYD15A2H (currently 218456 datasets, from 2002-07-04 to present - approximately 300 datasets every 08 days)
- MODIS_TERRA_MOD15A2H (currently 248647 datasets, from 2000-02-18 to present - approximately 300 datasets every 08 days)
- MODIS_TERRA_MOD14A2 (currently 255161 datasets, from 2000-02-18 to present - approximately 300 datasets every 08 days)
- MODIS_AQUA_MYD14A2  (currently 223424 datasets, from 2002-07-04 to present - approximately 300 datasets every 08 days)

Due to the fact that granule queries now require collection identifiers, each collection has to be harvested with different harvesters. Depending on the configuration, a different collection will be collected. So, 12 different harvesters must be created with the a configuration similar to:
{
  "collection": "MYD13Q1",
  "start_date": "2002-07-04T00:00:00Z",
  "make_private": false
}

### Features:

- [X] includes new  features
- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes bugfix

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by davidcordeiro-deimos** [comment]
In order to harvest from the earliest product onwards, set the start_date in harvest configuration as "YYYY-MM-DDT00:00:00Z", the initial date for each collection was described previously

**Comment by davidcordeiro-deimos** [comment]
This harvester makes use of a new function added in GDACS harvester, which generates logs about the number of datasets harvested by job for each data source.

**Comment by davidcordeiro-deimos** [comment]
Data Source - NASA’s Earth Observation Data CMR

Description - The NASA’s Earth Observation Common Metadata Repository (CMR) is a high-performance, high-quality, continuously evolving metadata system that catalogs Earth Science data and associated service metadata records. These metadata records are registered, modified, discovered, and accessed through programmatic interfaces leveraging standard protocols and APIs. 

Image: https://earthdata.nasa.gov/img/earthdata-fb-image.jpg

**Comment by georgiana-b** [comment]
Hey @davidcordeiro-deimos!  
I wanted to try out this harvester today and I noticed the [entry point for it is 
missing](https://github.com/NextGeoss/ckanext-nextgeossharvest/blob/modis/setup.py#L89). Could you add it please?

**Comment by joaandrade** [comment]
Hi @georgiana-b, i've added the missing entry point.
The build is failing for a reason unrelated to this PR, it is failing in a Sentinel Harvester test.

File: setup.py [hunk 1]
```diff
@@ -87,6 +87,7 @@
         probav=ckanext.nextgeossharvest.harvesters:PROBAVHarvester
         glass_lai=ckanext.nextgeossharvest.harvesters:GLASS_LAI_Harvester
         plan4all=ckanext.nextgeossharvest.harvesters:Plan4AllHarvester
+	modis=ckanext.nextgeossharvest.harvesters:MODISHarvester
```
Comment (georgiana-b) [inline setup.py]: @joaandrade I think you have to import it in [`harvesters` package](https://github.com/NextGeoss/ckanext-nextgeossharvest/blob/096a463a136d4e60b3cf365e66253c1585e2f25e/ckanext/nextgeossharvest/harvesters/__init__.py#L8) first before trying to retrieve it. Right now I get an import error:

```
ckan-dev_1    |   File "/usr/bin/paster", line 11, in <module>
ckan-dev_1    |     load_entry_point('PasteScript==2.0.2', 'console_scripts', 'paster')()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 102, in run
ckan-dev_1    |     invoke(command, command_name, options, args[1:])
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 141, in invoke
ckan-dev_1    |     exit_code = runner.run(args)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 236, in run
ckan-dev_1    |     result = self.command()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/serve.py", line 284, in command
ckan-dev_1    |     relative_to=base, global_conf=vars)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/serve.py", line 329, in loadapp
ckan-dev_1    |     **kw)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 247, in loadapp
ckan-dev_1    |     return loadobj(APP, uri, name=name, **kw)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 272, in loadobj
ckan-dev_1    |     return context.create()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 710, in create
ckan-dev_1    |     return self.object_type.invoke(self)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/deploy/loadwsgi.py", line 146, in invoke
ckan-dev_1    |     return fix_call(context.object, context.global_conf, **context.local_conf)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/deploy/util.py", line 55, in fix_call
ckan-dev_1    |     val = callable(*args, **kw)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/config/middleware/__init__.py", line 46, in make_app
ckan-dev_1    |     load_environment(conf, app_conf)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/config/environment.py", line 99, in load_environment
ckan-dev_1    |     p.load_all()
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/plugins/core.py", line 139, in load_all
ckan-dev_1    |     load(*plugins)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/plugins/core.py", line 153, in load
ckan-dev_1    |     service = _get_service(plugin)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/plugins/core.py", line 255, in _get_service
ckan-dev_1    |     return plugin.load()(name=plugin_name)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2411, in load
ckan-dev_1    |     return self.resolve()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2421, in resolve
ckan-dev_1    |     raise ImportError(str(exc))
ckan-dev_1    | ImportError: 'module' object has no attribute 'MODISHarvester'
```

**Comment by joaandrade** [comment]
@georgiana-b you're right. I've added the import to the __init__.py file

**Pull request merged**